enhancements/bpfd-on-ocp.md

```diff
@@ -0,0 +1,424 @@
+---
+title: bpfd-on-ocp
+authors:
+  - "@anfredette"
+  - "@astoycos"
+  - "@Billy99"
+  - "@dave-tucker"
+reviewers:
+  - "@dhellmann"
+  - "@bparees"
+  - "@deads2k"
+  - "@msherif1234"
+  - "@jotak"
+  - "@erthalion"
+  - "@danwinship"
+approvers:
+  - "@knobunc"
+api-approvers: 
+  - "@knobunc"
+creation-date: 2023-02-21
+last-updated: 2023-03-14
+tracking-link:
+see-also:
+replaces:
+superseded-by:
+---
+
+# BPF Support for OCP with bpfd
+
+## Summary
+
+This enhancement proposes the following:
+- That the bpfd operator be included in OCP as a foundational operator.
+- That the bpfd operator be used by internal Red Hat OCP applications and
+  components for the loading and management of required BPF programs.
+- That bpfd be recommended and supported for use by OCP partners and users to
+  load and manage their BPF programs when running on OCP.
+
+## Motivation
+
+BPF is a powerful technology that OCP developers and customers alike are
+increasingly trying to leverage. However, there are currently many challenges
+with using BPF programs in Kubernetes-based systems.
+
+[Enhancement Proposal
+1133](https://github.com/openshift/enhancements/pull/1133), “Guidelines for the
+Use of eBPF in OCP”, opened by Dan Winship last year, resulted in many
+questions/concerns such as:
+
+- How should multiple BPF programs share the same hook points? 
+- How should BPF be utilized in OCP by the infrastructure team AND customers
+  alike? 
+- How can multiple infra operators manage BPF programs without duplicating
+  effort?
+- How does BPF in OCP affect cluster debugging?
+
+Additional challenges with the proliferation of BPF in OCP include security,
+visibility, debuggability, and lifecycle management of BPF programs in general.
+
+Finally, without a common infrastructure for loading and managing BPF programs,
+each developer needs to solve these problems independently, which results in
+duplicated code and functionality.
+
+[Bpfd](https://bpfd.netlify.app) is a system daemon that supports, loading,
```

Comment:
There are 4 separate license files in https://github.com/redhat-et/bpfd.

Which one applies? The badge on the readme seems to imply Apache 2.0?

Reply:
The licensing description is here: https://github.com/redhat-et/bpfd#license
We'll update the badge accordingly.